HW2/P5/physics.pyx

```diff
@@ -69,17 +78,39 @@ cdef void sub_update(FLOAT[:, ::1] XY,
     ############################################################
     # SUBPROBLEM 2: use the grid values to reduce the number of other
     # objects to check for collisions.
-    for j in range(i + 1, count):
-        XY2 = &(XY[j, 0])
-        V2 = &(V[j, 0])
-        if overlapping(XY1, XY2, R):
-            # SUBPROBLEM 4: Add locking
-            if not moving_apart(XY1, V1, XY2, V2):
-                collide(XY1, V1, XY2, V2)
-
-            # give a slight impulse to help separate them
-            for dim in range(2):
-                V2[dim] += eps * (XY2[dim] - XY1[dim])
+
+    # Number of grid squares across grid
+    size = <unsigned int> (1+(1.0/grid_spacing))
+
+    # Get x and y grid coordinates
+    x = <unsigned int> (XY1[0]/grid_spacing)
+    y = <unsigned int> (XY1[1]/grid_spacing)
+
+    # Check grid spaces within 2 squares up or right
```

Comment:
Why are you checking only up or right and not down or left? I get slower performance than you because Im checking also those parts. Ill be curious to know if you dont miss any collisions with this version.